### PR TITLE
fix: mem cache out of range panic caused by overflow

### PR DESCRIPTION
**Pull request by jayantxie**
fix https://github.com/go-delve/delve/issues/3760

### CODE REVIEW

**Comment by aarzilli** [comment]
See failing test (TestIssue554 in proc_general_test.go).

**Comment by jayantxie** [comment]
> See failing test (TestIssue554 in proc_general_test.go).

Fixed.

**Review by aarzilli — APPROVED** [review]

File: pkg/proc/mem.go [hunk 1]
```diff
@@ -35,7 +35,12 @@ type memCache struct {
 }
 
 func (m *memCache) contains(addr uint64, size int) bool {
-	return addr >= m.cacheAddr && addr <= (m.cacheAddr+uint64(len(m.cache)-size))
+	end := addr + uint64(size)
+	if end < addr {
```
Comment (derekparker) [inline pkg/proc/mem.go:39]: This branch is never exercised by any test. Can you add a case for it?

**Review by derekparker — CHANGES_REQUESTED** [review]

File: pkg/proc/mem.go (hunk 1 above)
Comment (jayantxie) [reply pkg/proc/mem.go:39]: Done, please take a look again.

**Review by derekparker — APPROVED** [review]
LGTM

**Pull request merged**